src/components/DialogHeader/DialogHeader.tsx

```diff
@@ -0,0 +1,60 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { ReactElement } from 'react';
+import { merge } from '@utilities/merge';
+import { IconCross } from '@foundation/Icon';
+import { Badge, BadgeProps } from '@components/Badge';
+
+export type DialogHeaderProps = {
+    title: string;
+    size?: DialogHeaderSize;
+    collapseBottom?: boolean;
+    close?: boolean;
```

Comment:
Having `close` and `onClose` like this is not good IMO. 

Reason: What if the user puts close to true but does not provide the `onClose` function? 

What you should do is either only have onClose which is optional and a display button once that is provided, or you need to tightly couple those two props in such a way that if a user passes `close={true}` then `onClose` should be mandatory.